pathtools/fs.go

```diff
@@ -220,8 +224,16 @@ func (m *mockFs) Exists(name string) (bool, bool, error) {
 }
 
 func (m *mockFs) IsDir(name string) (bool, error) {
+	origName := name
 	name = filepath.Clean(name)
 	name = m.followSymlinks(name)
+
+	dir := filepath.Dir(name)
+	dir = m.followSymlinks(dir)
```

Comment:
followSymlinks from 229 should have already done this, right?

Reply:
I'll move this above m.followSymlinks(name).  It could be different if dir is a symlink and name is a symlink, for example:

foo/a -> a
bar -> foo

filepath.Dir(m.followSymlinks("bar/a")) will give "."
m.followSymlinks(filepath.Dir("bar/a")) will give "foo"

I'll try to add a test that covers this.

Reply:
Fixed, and added new tests that would have incorrectly reported ENOENT but now report ENOTDIR.